integration.js

```diff
@@ -79,20 +82,25 @@ function doLookup(entities, options, cb) {
               try {
                 file._icon = mimeTypes[file.mimeType] ? mimeTypes[file.mimeType] : DEFAULT_FILE_ICON;
                 file._typeForUrl = getTypeForUrl(file);
-                if (file.hasThumbnail) {
+                if (file.hasThumbnail && options.shouldDisplayFileThumbnails) {
                   file._thumbnailBase64 = await downloadThumbnail(tokens.access_token, file.thumbnailLink);
                 }
+                if (options.shouldGetFileContent) {
+                  file._content = await getFileContent(drive, file);
```

Comment:
@penwoodjon I think we need to move the file content download to `onMessage` so we only load the content when a user expands the drop down in the details block.  Otherwise we're potentially pulling down a lot of data at once.  I was seeing noticeable slow downs and even some errors when testing this out on the springbud server.

Reply:
Makes sense!  The searching, possibility of no content being returned, and way we go about actually doing this could take a bit of time though with other things in the pipeline. That's why I initially put this one behind a user option that is defaulted to false 😅.  I think I'll just have the first 1-3 files initially load, assuming no major slow downs, then have the remainder be onMessage.  That way the search feature can still do it's magic in terms of automatically expanding the file contents.

Reply:
@sarus Added the changes so all files are obtained with onMessage, and I was able to get the searching working as well! Let me know what you think and I'll add any changes you think are needed!